packages/std/src/types.rs

```diff
@@ -14,14 +14,18 @@ pub struct Env {
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, JsonSchema)]
 pub struct BlockInfo {
     pub height: u64,
-    // time is seconds since epoch begin (Jan. 1, 1970)
+    /// Absolute time of the block creation in seconds since the UNIX epoch (00:00:00 on 1970-01-01 UTC).
+    ///
+    /// The source of this is the [BFT Time in Tendermint](https://docs.tendermint.com/master/spec/consensus/bft-time.html),
+    /// converted from millisecond to second precision in an undefined way.
```

Comment:
"undefined way"

can we just define it to round down (drop off the milliseconds)?

Reply:
> can we just define it to round down (drop off the milliseconds)?

If you show me the explicit code where this is documented, implemented and tested, yes.

Right now we use [time.Time.Unix()](https://golang.org/pkg/time/#Time.Unix), which does not define what happens to fractional seconds. Also I don't know how the conversion from Tendermint's millisecond precision to the SDK's time.Time works.

Reply:
Okay, leave as unspecified. I have no desire to dig in that code now.

I am quite sure the `Unix()` time truncates milliseconds, but I cannot prove it.

Reply:
Yeah same here. I stopped tracing the code down when bit shifting started in the implementation.

Reply:
Couldn't drop it:

https://golang.org/src/time/time.go?s=35246:35272#L167

A bit confusing with the bit shifts, but if you read the docs above, it is clear that all the bits that are not encoding seconds are truncated.